doc/audit/security_auditing_HLD.md

```diff
@@ -0,0 +1,513 @@
+key                          = AUDIT|config                 ; Audit configuration settings
+; field                      = value
+enable                       = boolean                      ; Indicates whether security auditing is enabled (true/false)
+RULESET                      = list                         ; List of audit rule sets
+name                         = 1*255VCHAR                   ; Name of the audit rule set
+rule                         = 1*255VCHAR                   ; Audit rule definition in auditd format
+```
+
+##### 3.3.1.2 Config DB JSON Sample
+The predefined list of rules in section 3.2 will be enabled as default. Example of how the audit rules might be represented in JSON format within the Config DB
+```
+{
+    "AUDIT": {
+        "config": {
+            "enable": "true",
+            "RULESET": [
+                {
+                    "name": "file_deletion",
+                    "rules": [
+                        "-a exit,always -F arch=b64 -S unlink -S unlinkat -F key=file_deletion",
+                        "-a exit,always -F arch=b32 -S unlink -S unlinkat -F key=file_deletion"
+                    ]
+                },
+                {
+                    "name": "dns_changes",
+                    "rules": [
+                        "-w /etc/resolv.conf -p wa -k dns_changes"
+                    ]
+                }
+            ]
+        }
+    }
+}
+```
+
+##### 3.3.1.3 Redis Entries Sample
+Once the AUDIT table is populated in the Config DB, the corresponding entries can be viewed in Redis. Below are example Redis commands and outputs
+```
+127.0.0.1:6379[4]> keys AUDIT|config
+1) "AUDIT|config"
+
+127.0.0.1:6379[4]> hgetall AUDIT|config
+1) "enable"
+2) "true"
+3) "RULESET|file_deletion|1"
+4) "-a exit,always -F arch=b64 -S unlink -S unlinkat -F key=file_deletion"
+5) "RULESET|file_deletion|2"
+6) "-a exit,always -F arch=b32 -S unlink -S unlinkat -F key=file_deletion"
+7) "RULESET|dns_changes|1"
+8) "-w /etc/resolv.conf -p wa -k dns_changes"
+```
+
+#### 3.3.2 YANG model
+New YANG model `sonic-audit.yang` will be added.
+```
+module sonic-audit {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-audit";
+
+    prefix sonic-audit;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "AUDIT YANG Module for SONiC OS";
+
+    revision 2024-08-12 {
+        description "First Revision";
+    }
+
+    container sonic-audit {
+
+        container AUDIT {
+
+            description "AUDIT part of config_db.json";
+
+            container config {
+
+                leaf enable {
+                    description "This configuration identicates whether enable audit";
+                    type stypes:boolean_type;
+                    default "true";
+                }
+
+                list RULESET {
+                    key "name";
+                    description "List of audit rules";
+
+                    leaf name {
+                        type string;
```

Comment:
In schema you defined length, how about add length here to restrict?

Reply:
Updated